src/shared/tpm2-util.c

```diff
@@ -311,55 +321,222 @@ static int tpm2_credit_random(ESYS_CONTEXT *c) {
+                                .nameAlg = TPM2_ALG_SHA256,
+                                .objectAttributes = TPMA_OBJECT_FIXEDTPM|TPMA_OBJECT_FIXEDPARENT|TPMA_OBJECT_SENSITIVEDATAORIGIN|TPMA_OBJECT_RESTRICTED|TPMA_OBJECT_DECRYPT|TPMA_OBJECT_USERWITHAUTH|TPMA_OBJECT_NODA,
+                                .parameters.eccDetail = {
+                                        .symmetric = {
+                                                .algorithm = TPM2_ALG_AES,
+                                                .keyBits.aes = 128,
+                                                .mode.aes = TPM2_ALG_CFB,
+                                        },
+                                        .scheme.scheme = TPM2_ALG_NULL,
+                                        .curveID = TPM2_ECC_NIST_P256,
+                                        .kdf.scheme = TPM2_ALG_NULL,
                                 },
-                                .scheme.scheme = TPM2_ALG_NULL,
-                                .keyBits = 2048,
                         },
                 },
         };
 
+        /* make sure if someone uses something other than true false we don't explode */
+        return &templ[((use_srk_model & 0x1) << 1) + (is_ecc & 0x1)];
+}
+
+#define SRK_HANDLE 0x81000001
+/*
+ * See tpm2-tss bug: https://github.com/tpm2-software/tpm2-tss/issues/2553
+ * Keep it simple, just use the raw numerics.
+ */
+#define TPM2_HR_PERSISTENT_FIRST_GOOD  0x81000000
+
+static int tpm2_get_srk(
+                ESYS_CONTEXT *c,
+                ESYS_TR *ret_primary,
+                TPMI_ALG_PUBLIC alg,
+                TPMI_ALG_PUBLIC *ret_alg) {
+
+        TPMI_YES_NO more_data;
+        ESYS_TR primary = ESYS_TR_NONE;
+        _cleanup_(Esys_Freep) TPMS_CAPABILITY_DATA *cap_data = NULL;
+
+        TSS2_RC rc = sym_Esys_GetCapability(c,
+                        ESYS_TR_NONE,
+                        ESYS_TR_NONE,
+                        ESYS_TR_NONE,
+                        TPM2_CAP_HANDLES,
+                        TPM2_HR_PERSISTENT_FIRST_GOOD,
+                        TPM2_MAX_CAP_HANDLES,
+                        &more_data,
+                        &cap_data);
+        if (rc != TSS2_RC_SUCCESS)
+                return log_error_errno(SYNTHETIC_ERRNO(ENOTRECOVERABLE),
+                                       "Failed to enumerate handles searching for SRK: %s",
+                                       sym_Tss2_RC_Decode(rc));
+
+        for (UINT32 i=0; i < cap_data->data.handles.count; i++) {
+                TPM2_HANDLE handle = cap_data->data.handles.handle[i];
+                if (handle != SRK_HANDLE)
+                        continue;
+
+                log_debug("Found SRK on TPM.");
+
+                rc = sym_Esys_TR_FromTPMPublic(c,
+                        handle,
+                        ESYS_TR_NONE,
+                        ESYS_TR_NONE,
+                        ESYS_TR_NONE,
+                        &primary);
```

Comment:
as per coding style: if we break function parameter lists, please indent to double the usual indentation, i.e. 16 spaces.